### ISSUE TITLE
Merge table failed when drag&drop a shx file

### ISSUE THREAD

**Issue by GeoDaTester** [GeoDaCenter/geoda#1225]
From: jeremypj@umich.edu
 Details:
os: 1-10-12
vs: 1-12-0-0
/Applications/GeoDa.app/Contents/MacOS/../Resources/logger.txt
 GdaFrame::ShowOpenDatasourceDlg()
ConnectDatasourceDlg::InitSamplePanel()
Check auto update:
AutoUpdate::CheckUpdate()
AutoUpdate::GetCheckList()
AutoUpdate::ReadUrlContent()
Click GdaFrame::OnOpenProject
Entering ConnectDatasourceDlg::OnOkClick
ConnectDatasourceDlg::CreateDataSource()
Open Datasource:/Users/jeremiahjohnson/Desktop/Counties_v17a/Counties.shp
Open Layer:Counties
Entering ConnectDatasourceDlg::SaveRecentDataSource
Exiting ConnectDatasourceDlg::SaveRecentDataSource
Exiting ConnectDatasourceDlg::OnOkClick
Entering Project::Project (new project)
Project::CommonProjectInit()
Entering Project::InitFromOgrLayer
Datasource name:
/Users/jeremiahjohnson/Desktop/Counties_v17a/Counties.shp
Project::SetupEncoding()
UTF-8
Project::GetSpatialReference()
Exiting Project::Project
Click GdaFrame::InitWithProject()
Open TableFrame.
Open MapFrame.
MapCanvas::MapCanvas()
MapCanvas::ChangeMapType()
MapCanvas::VarInfoAttributeChange()
MapCanvas::CreateAndUpdateCategories()
Click GdaFrame::OnCloseProjectEvt
Project::GetProjectFullPath()
In TableFrame::OnClose and actually closing.
Project::CleanupPairsHLState()
Entering Project::~Project
Exiting Project::~Project
MapCanvas::~MapCanvas()
Click GdaFrame::OnOpenProject
Click GdaFrame::OnNewProject
 GdaFrame::ShowOpenDatasourceDlg()
ConnectDatasourceDlg::InitSamplePanel()
Entering ConnectDatasourceDlg::OnOkClick
Open CsvFieldConfDlg.
CsvFieldConfDlg::PrereadCSV()
CsvFieldConfDlg::UpdateFieldGrid()
CsvFieldConfDlg::ReadCSVT()
CsvFieldConfDlg::UpdatePreviewGrid()
CsvFieldConfDlg::UpdateXYcombox()
CsvFieldConfDlg::OnOkClick()
CsvFieldConfDlg::WriteCSVT()
ConnectDatasourceDlg::CreateDataSource()
Open Datasource:/Users/jeremiahjohnson/Desktop/EQIDATA_ALL_DOMAINS_2014MARCH11.CSV
Open Layer:EQIDATA_ALL_DOMAINS_2014MARCH11
Entering ConnectDatasourceDlg::SaveRecentDataSource
Exiting ConnectDatasourceDlg::SaveRecentDataSource
Exiting ConnectDatasourceDlg::OnOkClick
Entering Project::Project (new project)
Project::CommonProjectInit()
Entering Project::InitFromOgrLayer
Datasource name:
/Users/jeremiahjohnson/Desktop/EQIDATA_ALL_DOMAINS_2014MARCH11.CSV
Project::GetSpatialReference()
Exiting Project::Project
Click GdaFrame::InitWithProject()
Open TableFrame.
In TableFrame::OnActivate
Click GdaFrame::OnCloseProjectEvt
Project::GetProjectFullPath()
In TableFrame::OnClose and actually closing.
Project::CleanupPairsHLState()
Entering Project::~Project
Exiting Project::~Project
Click GdaFrame::OnNewProject
 GdaFrame::ShowOpenDatasourceDlg()
ConnectDatasourceDlg::InitSamplePanel()
Entering ConnectDatasourceDlg::SaveRecentDataSource
Exiting ConnectDatasourceDlg::SaveRecentDataSource
Entering Project::Project (new project)
Project::CommonProjectInit()
Entering Project::InitFromOgrLayer
Datasource name:
/Users/jeremiahjohnson/Desktop/Counties_v17a/Counties.shp
Click GdaFrame::OnNewProject
 GdaFrame::ShowOpenDatasourceDlg()
ConnectDatasourceDlg::InitSamplePanel()
Entering ConnectDatasourceDlg::OnOkClick
ConnectDatasourceDlg::CreateDataSource()
Open Datasource:/Users/jeremiahjohnson/Desktop/Counties_v17a/Counties.shx
Open Layer:Counties
Entering ConnectDatasourceDlg::SaveRecentDataSource
Exiting ConnectDatasourceDlg::SaveRecentDataSource
Exiting ConnectDatasourceDlg::OnOkClick
Entering Project::Project (new project)
Project::CommonProjectInit()
Entering Project::InitFromOgrLayer
Datasource name:
/Users/jeremiahjohnson/Desktop/Counties_v17a/Counties.shx
Click GdaFrame::OnNewProject
 GdaFrame::ShowOpenDatasourceDlg()
ConnectDatasourceDlg::InitSamplePanel()
Entering ConnectDatasourceDlg::OnOkClick
ConnectDatasourceDlg::CreateDataSource()
Open Datasource:/Users/jeremiahjohnson/Desktop/Counties.shp
Open Layer:Counties
Entering ConnectDatasourceDlg::SaveRecentDataSource
Exiting ConnectDatasourceDlg::SaveRecentDataSource
Exiting ConnectDatasourceDlg::OnOkClick
Entering Project::Project (new project)
Project::CommonProjectInit()
Entering Project::InitFromOgrLayer
Datasource name:
/Users/jeremiahjohnson/Desktop/Counties.shp
Click GdaFrame::OnOpenProject
Click GdaFrame::OnNewProject
 GdaFrame::ShowOpenDatasourceDlg()
ConnectDatasourceDlg::InitSamplePanel()
Entering ConnectDatasourceDlg::OnOkClick
ConnectDatasourceDlg::CreateDataSource()
Open Datasource:/Users/jeremiahjohnson/Desktop/Counties_v17a/Counties.shp
Open Layer:Counties
Entering ConnectDatasourceDlg::SaveRecentDataSource
Exiting ConnectDatasourceDlg::SaveRecentDataSource
Exiting ConnectDatasourceDlg::OnOkClick
Entering Project::Project (new project)
Project::CommonProjectInit()
Entering Project::InitFromOgrLayer
Datasource name:
/Users/jeremiahjohnson/Desktop/Counties_v17a/Counties.shp
Project::SetupEncoding()
UTF-8
Project::GetSpatialReference()
Exiting Project::Project
Click GdaFrame::InitWithProject()
Open TableFrame.
Open MapFrame.
MapCanvas::MapCanvas()
MapCanvas::ChangeMapType()
MapCanvas::VarInfoAttributeChange()
MapCanvas::CreateAndUpdateCategories()
Open MergeTableDlg.
Entering MergeTableDlg::OnOpenClick()
Entering MergeTableDlg::OnOpenClick()
Entering ConnectDatasourceDlg::OnOkClick
ConnectDatasourceDlg::CreateDataSource()
Open Datasource:/Users/jeremiahjohnson/Desktop/Counties_v17a/Counties.shp
Open Layer:Counties
Entering ConnectDatasourceDlg::SaveRecentDataSource
Exiting ConnectDatasourceDlg::SaveRecentDataSource
Exiting ConnectDatasourceDlg::OnOkClick
ds:/Users/jeremiahjohnson/Desktop/Counties_v17a/Counties.shp layer: Counties
Entering ConnectDatasourceDlg::OnOkClick
ConnectDatasourceDlg::CreateDataSource()
Open Datasource:/Users/jeremiahjohnson/Desktop/EQIDATA_ALL_DOMAINS_2014MARCH11.CSV
Open Layer:EQIDATA_ALL_DOMAINS_2014MARCH11
Entering ConnectDatasourceDlg::SaveRecentDataSource
Exiting ConnectDatasourceDlg::SaveRecentDataSource
Exiting ConnectDatasourceDlg::OnOkClick

**Comment by lixun910** [comment]
@YidanJ-Wu @jkoschinsky 
I was able to find the datasource online:
[Counties_v17a.zip](https://github.com/GeoDaCenter/geoda/files/1412839/Counties_v17a.zip)

Csv file: https://edg.epa.gov/data/Public/ORD/NHEERL/EQI/Eqidata_all_domains_2014March11.csv

**Comment by lixun910** [comment]
Here are the steps:
1. Drag and drop counties.shx to GeoDa
2. Table->Merge, click open dialog button, select the above csv file
3. One can't run Merge since the "Field Name Correction" dialog will always show "invalid field name"

**Comment by YidanJ-Wu** [comment]
Windows 8 1.12.1.29
I dragged and dropped counties.shx and got this error message: 
![image](https://user-images.githubusercontent.com/22801047/32024987-16a9dfdc-b9a4-11e7-8b3a-dfa099a9a53c.png)
next, I dragged and dropped counties.shp
Got this field name correction dialog: 
![image](https://user-images.githubusercontent.com/22801047/32025287-4079b9d0-b9a5-11e7-94b1-3d46cafb2deb.png)
Clicked OK, got this warning because the field selected had duplicates. 

I tried Merge again, this time by record order. Got the same field name correction dialog, clicked OK, and the merge was successful.
